jp2_pc/Source/Trespass/saveload.cpp

```diff
@@ -54,13 +54,14 @@ void AddToLoadGameVector(SAVEGAMEINFO * psgi, std::vector<SAVEGAMEINFO> * pInfo)
     std::vector<SAVEGAMEINFO>::iterator  ppinfo;
 
     for (ppinfo = pInfo->begin(), fEntered = false; 
-         ppinfo <= pInfo->end() && !fEntered; 
+         ppinfo != pInfo->end(); 
          ppinfo++)
     {
         if (CompareFileTime(&ppinfo->ft, &psgi->ft) == -1)
         {
             fEntered = true;
             pInfo->insert(ppinfo, *psgi);
+            break;
         }
     }
```

Comment:
After the insert we cannot do another iterator increment. We use `break` to leave the loop early, which would have ended anyway.